codalib/xsdatetime.py

```diff
@@ -0,0 +1,189 @@
+from datetime import datetime, tzinfo, timedelta
+
+from pytz import timezone, utc as utc_tz
+
+# Constants for time parsing/formatting
+# This is a stub
+XSDT_FMT = "%Y-%m-%dT%H:%M:%S"
+# This should never change
+XSDT_TZ_OFFSET = 19
+# The str value to use for default local timezone
+# In localization operations
+DEFAULT_LOCAL_TZ = "US/Central"
+
+
+class InvalidXSDateTime(Exception):
+    pass
+
+
+class XSDateTimezone(tzinfo):
+    """
+    Concrete subclass of tzinfo for making sense of timezone offsets.
+    Not really worried about localization here, just +/-HHMM
+    """
+
+    def __init__(self, hours=0, minutes=0, sign=1):
+        self.minutes = hours * 60 + minutes
+        self.minutes *= sign
+
+    def utcoffset(self, dt):
+        return timedelta(minutes=self.minutes)
+
+    def dst(self, dt):
+        return timedelta(0)
+
+
+def xsDateTime_parse(xdt_str, local_tz=None):
+    """
```

Comment:
I left that kwarg there so the function might be used with a timezone other than the current default. I made the default None so that the `DEFAULT_LOCAL_TZ` could be changed at runtime by a user of the library with `set_default_local_tz` after which calls to the parse function would use that timezone instead of the module's default. If the kwarg's default is set to the constant `DEFAULT_LOCAL_TZ`, the default is bound to the module's default value at the time it's imported and can't be changed by end-users of the library.

It's probably not something that will come up, but I thought it couldn't hurt. Definitely not a must, though, to be sure.